kore/src/Kore/Builtin/Int.hs

```diff
@@ -424,24 +428,58 @@ matchIntEqual =
             Monad.guard (hook2 == eqKey)
             & isJust
 
--- | Unification of Int values.
+data UnifyInt = UnifyInt
+    { int1, int2 :: !InternalInt
+    }
+
+-- | Matches two Int values that have equals sorts.
+matchInt ::
+    TermLike RewritingVariableName ->
+    TermLike RewritingVariableName ->
+    Maybe UnifyInt
+matchInt first second
+    | InternalInt_ int1 <- first
+      , InternalInt_ int2 <- second
+      , on (==) internalIntSort int1 int2 =
```

Comment:
I think this should be an assertion. Attempting to unify two things with different sorts doesn't make any sense.